src/opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -799,51 +831,127 @@ inline void keywordB( SummaryConfig::keyword_list& list,
   }
 }
 
-inline void keywordR2R( SummaryConfig::keyword_list& /* list */,
-                        const ParseContext& parseContext,
-                        ErrorGuard& errors,
-                        const DeckKeyword& keyword)
+inline std::string establishRegionContext(const DeckKeyword&       keyword,
+                                          const FieldPropsManager& field_props,
+                                          const ParseContext&      parseContext,
+                                          ErrorGuard&              errors,
+                                          SummaryConfigContext&    context)
 {
-    const auto& location = keyword.location();
-    std::string msg_fmt = "Region to region summary keyword {keyword} is ignored\n"
-                          "In {file} line {line}";
-    parseContext.handleError(ParseContext::SUMMARY_UNHANDLED_KEYWORD, msg_fmt, location, errors);
-}
+    auto region_name = std::string { "FIPNUM" };
 
+    if (keyword.name().size() > 5) {
+        region_name = "FIP" + keyword.name().substr(5, 3);
 
-inline void keywordR( SummaryConfig::keyword_list& list,
-                      SummaryConfigContext& context,
-                      const DeckKeyword& deck_keyword,
-                      const Schedule& schedule,
-                      const FieldPropsManager& field_props,
-                      const ParseContext& parseContext,
-                      ErrorGuard& errors ) {
+        if (! field_props.has_int(region_name)) {
+            const auto msg_fmt =
+                fmt::format("Problem with summary keyword {{keyword}}\n"
+                            "In {{file}} line {{line}}\n"
+                            "FIP region {} not defined in "
+                            "REGIONS section - {{keyword}} ignored", region_name);
 
-    auto keyword = deck_keyword.name();
-    if( is_region_to_region(keyword) ) {
-        keywordR2R( list, parseContext, errors, deck_keyword );
-        return;
-    }
-    std::string region_name = "FIPNUM";
-    if (keyword.size() > 5) {
-        region_name = "FIP" + keyword.substr(5,3);
-        if (!field_props.has_int(region_name)) {
-            std::string msg_fmt = fmt::format("Problem with summary keyword {{keyword}}\n"
-                                              "In {{file}} line {{line}}\n"
-                                              "FIP region {} not defined in REGIONS section - {{keyword}} ignored", region_name);
-            parseContext.handleError(ParseContext::SUMMARY_INVALID_FIPNUM, msg_fmt, deck_keyword.location(), errors);
-            return;
+            parseContext.handleError(ParseContext::SUMMARY_INVALID_FIPNUM,
+                                     msg_fmt, keyword.location(), errors);
+            return "";
```

Comment:
If we actually reach this `return "";` the following state will be broken?

Reply:
> If we actually reach this return ""; the following state will be broken?

Not really anymore than in [current master](https://github.com/OPM/opm-common/blob/4a8f432b1076912f42fd3f7ac17b1619d5a5685f/src/opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L835).  The primary difference is that this function is called from two sites and needs to signal failure differently than we currently do.